packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -263,9 +278,15 @@ export function useConnections({
     try {
       onConnected(connectionInfo, dataService);
 
-      // Update lastUsed date as now and save the connection.
-      connectionInfo.lastUsed = new Date();
-      await saveConnectionInfo(connectionInfo);
+      // if a connection has been saved already we only want to update the lastUsed
+      // attribute, otherwise we are going to save the entire connection info.
+      const connectionInfoToBeSaved =
+        (await connectionStorage.load(connectionInfo.id)) ?? connectionInfo;
```

Comment:
Do we need to reload the entire connection here? Would it make sense to check for `lastUsed` instead?

Reply:
we need to reload for the update, but we could do a check first, something like this?

``` js
const connectionInfoToBeSaved = connectionInfo.lastUsed && (await connectionStorage.load(connectionInfo.id)) ?? connectionInfo;
```

Reply:
Ah gotcha we are reloading here to pull in the original state of the connection then so the changes that are currently there do not override. This is probably a nicer way to do it than keeping around a clone of the original state of the connection then, I think both have implications if a user is doing things in another window.

Reply:
We are only replicating the old behavior here, i was surprised this is how the old connection model behaved.

Reply:
I think the suggestion I made for using `lastUsed` might not actually do anything then? Sorry!

Reply:
ah true, actually it would cause the entire thing to be saved on connect the first time, even if it was saved before, but just never used

Reply:
Changed it back. Let's wait for @addaleax before merging, It would be great to have some feedback, cause she wasn't fond of relaxing the validation too much. We can easily drop the `looseValidation` from `ensureWellFormedConnectionString`